paddle/fluid/imperative/jit/program_desc_tracer.cc

```diff
@@ -221,40 +220,47 @@ void ProgramDescTracer::InsertVarIfNotExist(
     const std::shared_ptr<VarBase> &new_var, bool is_input) {
   PADDLE_ENFORCE_NOT_NULL(new_var, platform::errors::InvalidArgument(
                                        "The variable to insert is NULL."));
-  if (vars_.count(new_var) != 0) return;
 
-  auto new_var_desc = new framework::VarDesc("");
-  vars_[new_var].reset(new_var_desc);
+  for (; vars_.count(new_var) == 0;) {
```

Comment:
How about just write `while (...)`?

Reply:
Sure. it's just a personal preference. When working with low level C code I would prefer `for` loop .